addons/crm_livechat/models/chatbot_script_step.py

```diff
@@ -90,6 +94,11 @@ def _process_step_create_lead_and_forward(self, discuss_channel):
                 lambda team: team.assignment_max
                 and lead.filtered_domain(literal_eval(team.assignment_domain or "[]"))
             )
+        if self.env.user.partner_id.company_id:
```

Comment:
it is probably a bad idea to create a lead, then change assignment right after that. You are going to have company / team / user mismatch, especially you globally try to fix all issues we had in base crm module.

My suggestion would be to have a common "create values" method, then create the lead in both methods but once. Don't create then change it, just create it as wanted directly.

Reply:
How could we apply the filter on `assignment_domain` on a non-created lead?
I don't think it's possible as it possibly depends on computed field that are correctly set only after the lead is created.

Reply:
Note: we likely never change assignment here, even though the code makes it look like it is possible.

Either the script has a forced team, and we'll use it almost no matter what (only exception: multi-company rule, in which case the team won't be previously assigned).

Or the script doesn't have a team, and it will be assigned once, at the end of this method, if an available user is found.

Reply:
Basically this new line could be moved into the `if not teams:` if that makes it more clear, as the same check is already done in `_process_step_create_lead`.